crates/rbuilder/src/live_builder/order_input/mod.rs

```diff
@@ -295,3 +298,54 @@ pub fn expand_path(path: PathBuf) -> eyre::Result<PathBuf> {
 
     Ok(PathBuf::from(shellexpand::full(path_str)?.into_owned()))
 }
+
+/// Performs maintenance operations on every new header by calling OrderPool::head_updated.
+/// Also calls some functions to generate metrics.
+async fn spawn_clean_orderpool_job<P>(
+    header_receiver: mpsc::Receiver<Header>,
+    provider_factory: P,
+    orderpool: Arc<Mutex<OrderPool>>,
+    global_cancellation: CancellationToken,
+) -> eyre::Result<JoinHandle<()>>
+where
+    P: StateProviderFactory + 'static,
+{
+    let mut header_receiver: mpsc::Receiver<Header> = header_receiver;
+
+    let handle = tokio::spawn(async move {
+        info!("Clean orderpool job: started");
+
+        while let Some(header) = header_receiver.recv().await {
+            let block_number = header.number;
+            set_current_block(block_number);
+            let state = match provider_factory.latest() {
+                Ok(state) => state,
+                Err(err) => {
+                    error!("Failed to get latest state: {}", err);
```

Comment:
please, do logs like this

`error!(?err, "Message")`

This is how its done in the rest of the codebase and it makes the errors appear as a field in the json logs instead of the message.